server/migrations/versions/973694cdbf49_add_a_sessions_table_as_we_move_.py

```diff
@@ -0,0 +1,31 @@
+# pylint: disable=invalid-name
+"""Add a sessions table as we move sessions to the backend.
+
+Revision ID: 973694cdbf49
+Revises: 74c579ae8555
+Create Date: 2023-01-27 00:45:14.646511+00:00
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '973694cdbf49'
+down_revision = '74c579ae8555'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.create_table('session',
+                    sa.Column('created_at', server.models.UTCDateTime(), nullable=False),
+                    sa.Column('updated_at', server.models.UTCDateTime(), nullable=False),
+                    sa.Column('id', sa.String(length=200), nullable=False),
+                    sa.Column('data', sa.JSON(), nullable=True),
```

Comment:
What exactly is this `data` object going to contain? I imagine at the very least a user type and user key:

https://github.com/votingworks/arlo/blob/6b5f18ee4782934cd48ac2e803e828375a833abc/server/auth/auth_helpers.py#L53-L54

If that understanding is correct, explicit fields for those values instead of a generic data object might make sense.

I've seen loosely defined sessions get overloaded with non-session data and misused as caches. While I'm confident that we wouldn't mistakenly do that, doesn't hurt to prevent that

Reply:
I agree with the risk you mention, but I think the pain of explicit database fields for sessions makes any added state management super painful. Even today, we have a number of other fields that go into the session, like the CSRF token, the OAuth session token for Auth0, the support-user separate auth layer. So we'd need to define at least 8-10 fields, and I suspect we'd still end up with an "other" JSON bucket in case we add another library that depends on Flask sessions in its own way.

Reply:
Got it! With that context, I agree that keeping things flexible makes sense